src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -1962,8 +2441,17 @@ namespace rascal {
       // Normalize and orthogonalize the radial coefficients
       radial_integral->finalize_coefficients(coefficients_center);
       if (compute_gradients) {
-        radial_integral->template finalize_coefficients_der<ThreeD>(
-            expansions_coefficients_gradient, center);
+        auto center_pair{center.get_atom_ii()};
+        auto & coefficients_center_pair_gradient{
+            expansions_coefficients_gradient[center_pair]};
+        radial_integral->template finalize_neighbour_derivative<ThreeD>(
+            coefficients_center_pair_gradient);
+        for (auto neigh : center.pairs()) {
+          auto & coefficients_neigh_gradient{
+              expansions_coefficients_gradient[neigh]};
+          radial_integral->template finalize_neighbour_derivative<ThreeD>(
+              coefficients_neigh_gradient);
+        }  // for (neigh : center)
```

Comment:
so using just `center.pairs_with_self_pair()` and removing the `get_atom_ii` specific code is not equivalent to what is happening here?